fusillade/clouddirectory.py

```diff
@@ -0,0 +1,807 @@
+                {
+                    'SchemaArn': self._schema,
+                    'FacetName': facet_name
+                },
+            ],
+            'ObjectAttributeList': object_attribute_list,
+            'ParentReference': {
+                'Selector': parent
+            },
+            'LinkName': name,
+        }
+        }
+
+    def batch_get_attributes(self, obj_ref, facet, attributes: typing.List[str]) -> typing.Dict[str, typing.Any]:
+        return {
+            'GetObjectAttributes': {
+                'ObjectReference': {
+                    'Selector': obj_ref
+                },
+                'SchemaFacet': {
+                    'SchemaArn': self._schema,
+                    'FacetName': facet
+                },
+                'AttributeNames': attributes
+            }
+        }
+
+    @staticmethod
+    def batch_attach_object(parent: str, child: str, name: str) -> typing.Dict[str, typing.Any]:
+        """
+        batch write operation
+        :param parent:
+        :param child:
+        :param name:
+        :return:
+        """
+        return {
+            'AttachObject': {
+                'ParentReference': {
+                    'Selector': parent
+                },
+                'ChildReference': {
+                    'Selector': child
+                },
+                'LinkName': name
+            }
+        }
+
+    @staticmethod
+    def batch_detach_object(parent: str, link_name: str) -> typing.Dict[str, typing.Any]:
+        return {'DetachObject': {
+            'ParentReference': {
+                'Selector': parent
+            },
+            'LinkName': link_name,
+        }}
+
+    @staticmethod
+    def batch_attach_policy(policy: str, object_ref: str) -> typing.Dict[str, typing.Any]:
+        """
+        batch write operation
+        :param policy:
+        :param object_ref:
+        :return:
+        """
+        return {
+            'AttachPolicy': {
+                'PolicyReference': {
+                    'Selector': policy
+                },
+                'ObjectReference': {
+                    'Selector': object_ref
+                }
+            }
+        }
+
+    def batch_write(self, operations: list) -> typing.Dict[str, typing.Any]:
+        return ad.batch_write(DirectoryArn=self._dir_arn, Operations=operations)
+
+    def batch_read(self, operations: typing.List[typing.Dict[str, typing.Any]]) -> typing.Dict[str, typing.Any]:
+        return ad.batch_read(DirectoryArn=self._dir_arn, Operations=operations)
+
+    @staticmethod
+    def get_obj_type_path(obj_type: str) -> str:
+        obj_type = obj_type.lower()
+        paths = dict(group='/Groups/',
+                     index='/Indices/',
+                     user='/Users/',
+                     policy='/Policies/',
+                     role='/Roles/')
+        return paths[obj_type]
+
+    def lookup_policy(self, object_id: str) -> typing.List[str]:
+        max_results = 3
```

Comment:
Why 3?  Maybe comment on min/max restrictions if tweaks are desired in the future.

Reply:
according to AWS tech support anything more than 3 will crash, so we'll just page it to be safe